pkg/virtctl/vmexport/vmexport.go

```diff
@@ -461,31 +607,48 @@ func waitForVirtualMachineExport(client kubecli.KubevirtClient, vmeInfo *VMExpor
 		}
 
 		if vmexport.Status.Phase != exportv1.Ready {
-			fmt.Printf("Waiting for VM Export %s status to be ready...\n", vmeInfo.Name)
+			fmt.Printf("waiting for VM Export %s status to be ready...\n", vmeInfo.Name)
 			return false, nil
 		}
 
-		if vmexport.Status.Links == nil || vmexport.Status.Links.External == nil {
-			fmt.Printf("Waiting for VM Export %s external links to be available...\n", vmeInfo.Name)
-			return false, nil
+		if exportURL == "" {
+			if vmexport.Status.Links == nil || vmexport.Status.Links.External == nil {
+				fmt.Printf("waiting for VM Export %s external links to be available...\n", vmeInfo.Name)
+				return false, nil
+			}
+		} else {
+			if vmexport.Status.Links == nil || vmexport.Status.Links.Internal == nil {
+				fmt.Printf("waiting for VM Export %s internal links to be available...\n", vmeInfo.Name)
+				return false, nil
+			}
 		}
-
-		fmt.Printf("Processing completed successfully\n")
+		fmt.Printf("processing completed successfully\n")
 		return true, nil
 	})
 
 	return err
 }
 
-// handleHTTPRequest generates the GET request with proper certificate handling
-func handleHTTPRequest(client kubecli.KubevirtClient, vmexport *exportv1.VirtualMachineExport, downloadUrl string, insecure bool) (*http.Response, error) {
+// HandleHTTPRequestFunc function used to handle http requests
+type HandleHTTPRequestFunc func(client kubecli.KubevirtClient, vmexport *exportv1.VirtualMachineExport, downloadUrl string, insecure bool, exportURL string, headers map[string]string) (*http.Response, error)
+
+// instance of function used to handle http requests
+var HandleHTTPRequest HandleHTTPRequestFunc = handleHTTPGetRequest
+
+// handleHTTPGetRequest generates the GET request with proper certificate handling
+func handleHTTPGetRequest(client kubecli.KubevirtClient, vmexport *exportv1.VirtualMachineExport, downloadUrl string, insecure bool, exportURL string, headers map[string]string) (*http.Response, error) {
 	token, err := getTokenFromSecret(client, vmexport)
 	if err != nil {
 		return nil, err
 	}
 
+	var cert string
 	// Create new certPool and append our external SSL certificate
-	cert := vmexport.Status.Links.External.Cert
+	if exportURL == "" {
+		cert = vmexport.Status.Links.External.Cert
+	} else {
+		cert = vmexport.Status.Links.Internal.Cert
```

Comment:
To actually make use of this cert (no --insecure) user is doing to have to configure dns somehow.  Seems unlikely but whatever

Reply:
Right, it is highly unlikely, but not impossible. Alternative is to not use it at all and make it auto insecure. I don't think this hurts anything because most likely `--insecure` will be used.